app/src/main/java/org/cru/godtools/ui/dashboard/tools/ToolsCategoryDataModel.kt

```diff
@@ -0,0 +1,38 @@
+package org.cru.godtools.ui.dashboard.tools
+
+import android.app.Application
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.map
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+import org.ccci.gto.android.common.androidx.lifecycle.combineWith
+import org.ccci.gto.android.common.db.Expression.Companion.constants
+import org.ccci.gto.android.common.db.Query
+import org.ccci.gto.android.common.db.getAsLiveData
+import org.cru.godtools.base.ui.util.getLocalStringFromToolCategory
+import org.cru.godtools.model.Tool
+import org.keynote.godtools.android.db.Contract
+import org.keynote.godtools.android.db.GodToolsDao
+
+@HiltViewModel
+class ToolsCategoryDataModel @Inject constructor(dao: GodToolsDao, context: Application) : AndroidViewModel(context) {
+
+    val selectedCategory = MutableLiveData<String?>(null)
+
+    private val allTools = Query.select<Tool>().where(
+        Contract.ToolTable.FIELD_TYPE.`in`(*constants(Tool.Type.TRACT, Tool.Type.ARTICLE, Tool.Type.CYOA))
+            .and(Contract.ToolTable.FIELD_HIDDEN.ne(true))
+    ).orderBy(Contract.ToolTable.COLUMN_DEFAULT_ORDER).getAsLiveData(dao)
+
+    val categories = allTools.map {
+        it.mapNotNull { tool -> tool.category }.distinct()
+    }
+
+    fun getFormattedCategory(category: String) = category.getLocalStringFromToolCategory(getApplication())
```

Comment:
you don't need this method, you can call the ModelUtils method directly from within data binding

Reply:
```

            <TextView
                android:layout_width="match_parent"
                android:layout_height="wrap_content"
                android:layout_marginStart="20dp"
                android:layout_marginTop="15dp"
                android:layout_marginBottom="15dp"
                android:text="@{ModelUtils.getLocalStringFromToolCategory(category, context)}"
                app:layout_constraintBottom_toBottomOf="parent"
                app:layout_constraintEnd_toEndOf="parent"
                app:layout_constraintStart_toStartOf="parent"
                app:layout_constraintTop_toTopOf="parent"
                tools:text="Category" />

```
@frett Whenever I add the code above to implement it in binding the app crashes.

Reply:
it's probably because category is null before you actually bind anything to the model

Reply:
and that method currently requires a non-null string

Reply:
@frett the way you asked me to switch the code below actually worked in the binding so disregard the comment above.